src/Stratis.Bitcoin/Connection/ConnectionManager.cs

```diff
@@ -404,17 +404,7 @@ private bool ShouldDisconnect(INetworkPeer peer)
             }
 
             // Don't disconnect if this peer is in -addnode or -connect.
-            bool isAddNodeOrConnect = false;
-            foreach (IPEndPoint addNodeEndPoint in this.ConnectionSettings.AddNode.Union(this.ConnectionSettings.Connect))
-            {
-                if (peer.PeerEndPoint.Address.Equals(addNodeEndPoint.Address))
-                {
-                    isAddNodeOrConnect = true;
-                    break;
-                }
-            }
-
-            if (isAddNodeOrConnect)
+            if (this.ConnectionSettings.AddNode.Union(this.ConnectionSettings.Connect).Any(ep => peer.PeerEndPoint.MatchIpOnly(ep)))
```

Comment:
@codingupastorm if you use `MatchIpOnly` here, it means that if a peer with the **same IP but different port** connects, and it is the connect / add node collection it will be disconnected. Do we care about the port number in this instance?

(just thinking out loud here)

Reply:
Ah sorry I just realized @quantumagi did this PR, sorry @codingupastorm .

Reply:
So given this scenario:

1> 100.0.0.1:3200 connects.
2> 100.0.0.1:3233 is already in the `-addnode` or `-connect`.
3> Then 100.0.0.1:3200 will not be allowed to connect.

Is this correct?

Reply:
@fassadlr , Note that I am only using `MatchIpOnly` to remain compatible with the legacy code which was only matching on ip address. If we can reach agreement that the legacy code was incorrect in that regard then we can include matching on port as well.

Reply:
@quantumagi OK I agree. I was actually just thinking out loud :)

Reply:
@quantumagi @fassadlr I believe the matching by IP only is deliberate, although I can't recall exactly why without searching through old commits.

Reply:
This looks correct to me, though I would think we already map to ipv6 everywhere, what is the scenarios that leads to this change?